### PR TITLE
Add an endpoint for batch updates

### PR DESCRIPTION
**Pull request by allysonjp715**
Sorry this got fairly large in the end!

### CODE REVIEW

File: data-serving/data-service/src/controllers/preprocessor.ts [hunk 1]
```diff
@@ -113,6 +113,46 @@ export const setBatchRevisionMetadata = async (
...
+            caseReference: 1,
+            revisionMetadata: 1,
+        })
+        .exec();
+
+    const metadataMap = new Map(
+        existingCases
+            .filter((c) => c && c.caseReference)
+            .map((c) => [
+                c.caseReference.sourceId + ':' + c.caseReference.sourceEntryId,
```
Comment (axmb) [inline data-serving/data-service/src/controllers/preprocessor.ts:145]: For this method, we aren't guaranteed to have unique sourceId/sourceEntryId combinations. Do we need the key of this metadataMap to be unique?

File: data-serving/data-service/test/controllers/case.test.ts [hunk 2]
```diff
@@ -555,6 +555,50 @@ describe('PUT', () => {
...
     });
+    it('update many items should return 200 OK', async () => {
+        const c = new Case(minimalCase);
+        await c.save();
+        const c2 = new Case(minimalCase);
+        await c2.save();
+        await new Case(minimalCase).save();
+
+        const newNotes = 'abc';
+        const newNotes2 = 'cba';
```
Comment (axmb) [inline data-serving/data-service/test/controllers/case.test.ts:566]: >cba

😲

File: data-serving/data-service/api/openapi.yaml [hunk 3]
```diff
@@ -229,6 +229,31 @@ paths:
...
+                - $ref: "#/components/schemas/CaseArray"
+              properties:
+                curator:
+                  $ref: "#/components/schemas/Curator"
+              required:
+                - curator
+      responses:
+        "200":
+          $ref: "#/components/responses/200BatchUpdateResponse"
+        "422":
```
[the remark this answers is not in the document]
Comment (allysonjp715) [reply data-serving/data-service/api/openapi.yaml:253]: Ah yes thanks.

File: data-serving/data-service/api/openapi.yaml [hunk 4]
```diff
@@ -229,6 +229,31 @@ paths:
           $ref: "#/components/responses/422"
         "500":
           $ref: "#/components/responses/500"
+  /cases/batchUpdate:
+    put:
```
[the remark this answers is not in the document]
Comment (allysonjp715) [reply data-serving/data-service/api/openapi.yaml]: Done.

File: data-serving/data-service/src/controllers/case.ts [hunk 5]
```diff
@@ -330,6 +330,40 @@ export const update = async (req: Request, res: Response): Promise<void> => {
...
+                        filter: {
+                            _id: c._id,
+                        },
+                        update: c,
+                    },
+                };
+            }),
+            { ordered: false },
+        );
+        res.json({ numModified: bulkWriteResult.modifiedCount });
```
[the remark this answers is not in the document]
Comment (allysonjp715) [reply data-serving/data-service/src/controllers/case.ts:360]: Hm the documentation doesn't seem to outline this case specifically. But [updateOne](https://docs.mongodb.com/manual/reference/method/db.collection.updateOne/#db.collection.updateOne) returns a `matchedCount` and a `modifiedCount` which suggests that it only increments modifiedCount if the document was *actually* updated.

**Comment by allysonjp715** [comment]
> This batch of code is an excellent update to our repository.

Why thank you!

File: data-serving/data-service/src/controllers/preprocessor.ts (hunk 1 above)
Comment (allysonjp715) [reply data-serving/data-service/src/controllers/preprocessor.ts]: Yes, good call. Switched to using ids instead since that is required in the body.

**Pull request merged**